### ISSUE TITLE
repeater fields issues in Gutenberg

### ISSUE THREAD

**Issue by andrejarh** [siteorigin/so-widgets-bundle#730]
I just tested custom widgets in Gutenberg and noticed PHP Notices and Warnings appearing with each widget, where repeater field was used. It seems like the repeater data is missing:
![screen shot 2018-08-16 at 11 46 42](https://user-images.githubusercontent.com/10053621/44202162-773e7080-a14b-11e8-9bd0-b5915ce4e92b.png)
<img width="989" alt="screen shot 2018-08-16 at 11 50 20" src="https://user-images.githubusercontent.com/10053621/44202174-7f96ab80-a14b-11e8-8e5a-105a114347a3.png">
![screen shot 2018-08-16 at 11 49 51](https://user-images.githubusercontent.com/10053621/44202176-81f90580-a14b-11e8-90ed-998f3ea91463.png)

The notices and warnings appear only in Gutenberg editor.

I used [debug bar plugin](https://wordpress.org/plugins/debug-bar/) for the notices.
The issue appears on every repeater field in all of my custom widgets.

I didn't find any practical issues with the Gutenberg though. Everything seems to work OK.

I can define an empty array, restructure the code and prevent such notices in my plugins and themes, but I think this is a kind of issue, that probably affects other users too. 

So if this is something that could be easily fixed on your part, it would be great. :)

**Comment by AlexGStapleton** [comment]
Does this issue occur when you're using siteorigin/so-widgets-bundle#726?

**Comment by andrejarh** [comment]
Yes, this issue occurs with the gutenberg-compat branch too.
Oddly enough, the error appears also if the widget is not used on the specific page at all. 
The error returned is for all the active widgets, regardless if they are placed in the gutenberg editor or not:
![screen shot 2018-08-16 at 12 43 45](https://user-images.githubusercontent.com/10053621/44204472-f84d3600-a152-11e8-9838-2e7521c0103a.png)

**Comment by AlexGStapleton** [comment]
Does this issue only occur with custom widgets or does this issue occur with any of the default SiteOrigin Widgets Bundle widgets? I ask as I've run some tests and I can't seem to replicate this issue.

**Comment by andrejarh** [comment]
No, it doesn't happen with bundle widgets.

I investigated this further, and found out, that the problem is in the template naming. if the widget template file has the filename *template*.php then the notices are shown. When I renamed the file to something else, the notices were gone. 
Out of curiosity I renamed it to *loop.php and the notice reoccurred. 

The right formulation for this issue should be:
**For some reason the files which have the "template" or "loop" in their filename can get initiated without context in gutenberg.** 

I do not have full understanding of the issue, but at least it is pinned down a bit and should be easy to replicate
- rename widget template filename so that it contains "template" in the filename and adjust the get_template_name() accordingly.

TLDR:
The solution for anyone experiencing this issue is to **not** use the names "loop" or "template" in the widget-template filenames.

**Comment by AlexGStapleton** [comment]
Hi andrejarh,

I noticed you closed this issue and I'm just wondering why. Did you find a cause for this issue that's separate to the SiteOrigin Widgets Bundle?
